clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -1030,10 +1025,7 @@ private void blockingConnect(String node) throws IOException {
 
     protected void blockingConnect(String node, InetSocketAddress serverAddr) throws IOException {
         selector.connect(node, serverAddr, BUFFER_SIZE, BUFFER_SIZE);
-        while (!selector.connected().contains(node))
-            selector.poll(10000L);
-        while (!selector.isChannelReady(node))
-            selector.poll(10000L);
+        NetworkTestUtils.waitForChannelReady(selector, node);
```

Comment:
I think we should replace the 2 while loop with:
`waitForChannelConnected` + `waitForChannelReady`.
This way, we can make sure the connection is done or not when exception thrown during `waitForChannelReady`, right?

Same comments to below similar replacement.

Reply:
Let me add the KafkaChannel's state machine diagram for clarity. 

Given that the TCP connection is already established after connect completes successfully (state: READY|AUTHENTICATE), I don't think it is necessary to wait for channel readiness, which would unnecessary delay this test in case of TLS/SASL authentications. Instead, it is necessary for all the other tests on this class.

![kafka-channel-states](https://user-images.githubusercontent.com/11456498/169981699-f0e7fc6c-985b-407a-80c7-8294643f40f8.jpg)

Reply:
@showuon I probably expressed that badly. I added `NetworkTestUtils.waitForChannelConnected()` for tests that don't need to use the channel and don't care about the current state, like `SelectorTest.testCloseOldestConnection()` (we can test idle connection expiration from every state).